ihmc-perception/src/test/java/us/ihmc/perception/cuda/ExampleCUDAKernel.java

```diff
@@ -0,0 +1,119 @@
+package us.ihmc.perception.cuda;
+
+import org.bytedeco.cuda.cudart.CUstream_st;
+import org.bytedeco.cuda.cudart.dim3;
+import org.bytedeco.cuda.global.cudart;
+import org.bytedeco.javacpp.FloatPointer;
+import us.ihmc.log.LogTools;
+
+import static org.bytedeco.cuda.global.cudart.*;
+
+/**
+ * This is a simple example of a kernel that adds two arrays together. The ways in which kernels can be run vary such that you can create all the JavaCPP Pointers
+ * inside a try-with-resources. The user can create more threads on the GPU. This example attempted to keep things really simple and avoided most of that overhead.
+ */
+public class ExampleCUDAKernel
+{
+
+   private static final String KERNEL_TO_ADD_THE_VALUES_OF_TWO_ARRAYS = """
+         extern "C"
+         
+         __global__
+         void add_arrays(int n, float *x, float *y)
+         {
+            int index = blockIdx.x * blockDim.x + threadIdx.x;
+            int stride = blockDim.x * gridDim.x;
+            for (int i = index; i < n; i += stride)
+               y[i] = x[i] + y[i];
+         }
+         """;
+
+   // Even if you don't create a stream, CUDA will create a default one for you, so we make one ourselves to handle things better
+   public ExampleCUDAKernel()
+   {
+      // Note this name does NOT have to match the name of the program, however for readability its ideal to have them match.
+      String userFriendlyNameOfProgram = "userFriendlyNameOfProgram";
+      // Note this name DOES have to match the name of the CUDA kernel you want to run
+      String kernelName = "add_arrays";
+
+      // We create a stream to synchronize the method calls that happen on the GPU.
+      CUstream_st stream = new CUstream_st();
```

Comment:
Is there a reason you don't use `CUDAStreamManager` here?

Reply:
So we can show the low level logic of creating and destroying CUDA streams

Reply:
I think if you do that, you shouldn't use CUDAProgram or CUDAKernel classes. The typical usage is to use `CUDAStreamManager`

Reply:
The CUDAStreamManager hides some of the stuff that needs to happen with the stream. In this example we wanted to expose that to the user.

Reply:
Can you explain? I think you should use `CUDAStreamManager` here because that's how a user-developer would use it elsewhere.

Reply:
The CUDAStreamManager hides the fact that you need to close the stream, and it doesn't allow me to add comments explaining what the stream does. We only have the one stream in this example so I didn't need the manager. In the next example we are using the manager

Reply:
Okay, fair enough